tests/trainer/test_data_loading.py

```diff
@@ -115,8 +116,8 @@ def on_train_start(self, *_) -> None:
     else:
         warn_str = "Consider setting accelerator=ddp"
 
-    trainer = Trainer(accelerator="ddp_spawn")
     with pytest.warns(UserWarning, match=warn_str), pytest.raises(SystemExit):
+        trainer = Trainer(accelerator="ddp_spawn")
```

Comment:
nit: what was wrong with initializing the trainer outside this context?

Reply:
I won't fail as the SystemExit is triggered within the `on_train_epoch_start ` of the Callback.